src/node_util.cc

```diff
@@ -38,6 +38,11 @@ using v8::Value;
 #undef V
 
 
+static void IsPrimitive(const FunctionCallbackInfo<Value>& args) {
+  args.GetReturnValue().Set(!args[0]->IsObject() && !args[0]->IsExternal());
```

Comment:
How did you come up with this check? Does it match what the v8 native `%IsPrimitive()` checks?